### DISCUSSION TITLE
TS Error while linking to custom components from structure

### DISCUSSION THREAD

**Discussion by tstikvoort** [sanity-io/sanity#3911] [Sanity Studio Support] [answered]
Hi, I am migrating to Sanity V3 and am really happy with the typescript support.
In my structure I want to link to a custom view / component (in this case a component where you can import and review data). 
This is the code I wrote in the structure:

```typescript
S.listItem()
    .title('Import collection')
    .child(S.view.component(ImportCollectionView).id('ImportCollection').title('Import collection'))
```

It works as expected to me, by I do get the following typescript error:
```
Argument of type 'ComponentViewBuilder' is not assignable to parameter of type 'UnserializedListItemChild'.
  Type 'ComponentViewBuilder' is missing the following properties from type 'ComponentBuilder': child, getChild, menuItems, getMenuItems, and 2 more.ts(2345)
```

How can I fix this?

**Comment by bryandowning** [comment]
Have you tried just passing the component to `.child`?

```
S.listItem()
    .title('Import collection')
    .child(ImportCollectionView)
```

**Reply by tstikvoort** [reply]
Interesting, haven't thought of it, but that doesn't work. Both runtime and TS error.

**Comment by wrux** [comment, marked as answer]
Looks like you are close, how about this:

```jsx
S.listItem()
  .title('Import collection')
  .child(S.component(ImportCollectionView).id('ImportCollection').title('Import collection')),
```

**Reply by tstikvoort** [reply]
Yes, that works! thanks
